### DISCUSSION TITLE
DOI-Zenodo badge broken?

### DISCUSSION THREAD

**Discussion by Julian-Hochhaus** [lmfit/lmfit-py#862] [General]
Hi all,
For me, the doi bagde does not work, most interestingly, even on zenodo, the version number of the newest version 1.2.0 is not correctly resolved.
So the specific link to v.1.2.0 seems broken:

..  image:: https://zenodo.org/badge/4185/lmfit/lmfit-py.svg
    :target: https://doi.org/10.5281/zenodo.7810964

in Markdown:
[![DOI](https://zenodo.org/badge/4185/lmfit/lmfit-py.svg)](https://doi.org/10.5281/zenodo.7810964)

However, changing the target to the concept doi, which is always resolved to the latest version, works fine:
.. image:: https://zenodo.org/badge/4185/lmfit/lmfit-py.svg
   :target: https://doi.org/10.5281/zenodo.598352
Markdown: [![DOI](https://zenodo.org/badge/4185/lmfit/lmfit-py.svg)](https://doi.org/10.5281/zenodo.598352)
The version in the Readme:
.. image:: https://zenodo.org/badge/4185/lmfit/lmfit-py.svg
   :target: https://zenodo.org/badge/latestdoi/4185/lmfit/lmfit-py
Markdown: [![DOI](https://zenodo.org/badge/4185/lmfit/lmfit-py.svg)](https://zenodo.org/badge/latestdoi/4185/lmfit/lmfit-py)
does not work for me either. 

Maybe there is something wrong on my side, however I can't find out, whats wrong here.

**Comment by newville** [comment]
@Julian-Hochhaus Thanks -- I had not noticed that, but I see that too.  That is, it is not just you ;)

That badge should be pointing to https://zenodo.org/record/7810964, which sort of makes it look like everything is OK at Zenodo, but something is missing at doi.org.   I reported that as an error at doi.org.

Both https://zenodo.org/record/598352 (and so also https://doi.org/10.5281/zenodo.598352) redirect to https://zenodo.org/record/7810964.  I don't really understand that - I thought each release was supposed to have a unique key.

I guess we could update the badges to the 598352 badges, but that seems weird....  I think maybe more investigation is needed.

**Reply by Julian-Hochhaus** [reply]
The 598352 DOI is the so called concept DOI. Which currently always links to the latest version. Zenodo plans to introduce a landing page in the future, so that users are able to cite a package as it whole. I don’t know how they plan to handle it in the future, however, I assume that they will introduce links, that always link to the latest version, like the concept DOI currently does.

Each version therefore still has its unique DOI.

According to the [FAQ:](https://help.zenodo.org/)

> Where does the Concept DOI resolve to?
Currently the Concept DOI resolves to the landing page of the latest version of your record. This is not fully correct, and in the future we will change this to create a landing page specifically representing the concept behind the record and all of its versions. `

**Reply by newville** [reply]
Thanks.  That "concept" link does sort of seem like what we should be using. We don't really intend to publish every version explicitly, but it is great to have DOIs that point to the actual code for each version.

Do you think we should change all the links and badges to point to the "concept," or should we wait for Zenodo to announce that as more official?

**Reply by Julian-Hochhaus** [reply]
I think lmfit should link the concept DOI, currently, the forward to the newest version is fine as well as at some time in the future, when a landing page is introduced by zenodo, its still the best option. 

I will monitor announcements by zenodo and will come back to you if something happens.
